### PR TITLE
PHP example of adding a member to a list

### PR DESCRIPTION
**Pull request by jserrao**
Felt like closing this big gap in your documentation. I'm sure you guys could make a more canonical example than this but it works and would help many out, judging from what I'm seeing around the net, S.O., blogs, etc.

### CODE REVIEW

File: php/add-subscriber-to-list.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  // Setup destination URL
+  // Note that your API URL has that location subdomain up front
+  // It can vary, check the last 3 digits of your API key
+  $url = 'https://us5.api.mailchimp.com/3.0/lists/' . $list_id . '/members/';
+
+  // Encode data into a format that the add subscriber mailchimp end point is looking for
+  // Must include 'email_address' and 'status'
+  // Statuses: pending = they get an email; subscribed = they don't get an email
+  // Any custom fields go into the weirdly named 'merge_fields'
```
Comment (deezone) [inline php/add-subscriber-to-list.php]: It's not so "weirdly" as these are the field names used for merge functionality within templates used by both the Mailchimp admin UI and Mandrill.

Search for "merge_vars":
- https://mandrillapp.com/api/docs/messages.JSON.html#method=send-template 

I encourage you to remove your opinion comments from the PR. Fun to read but perhaps misleading.

Comment (jserrao) [reply php/add-subscriber-to-list.php]: I removed that language and tried to make the comments more readable. Check it out.

And I hear you from an architectural standpoint but think about dev UX here: most API users don't know what your internal architecture looks like. This took me like 45-60 minutes to figure out and I imagine it throws a lot of other people. All your API docs say for `merge_field`:

> An individual merge var and value for a member.

File: php/add-subscriber-to-list.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  /* ================
+   * DESTINATION URL
+   * Note: your API URL has a location subdomain at the front of the URL string
+   * It can vary depending on where you are in the world
+   * To determine yours, check the last 3 digits of your API key
+   * ================
+   */
+  $url = 'https://us5.api.mailchimp.com/3.0/lists/' . $list_id . '/members/';
+
+  /* ================
```
Comment (deezone) [inline php/add-subscriber-to-list.php:29]: For what it's worth, while not a part of the PSR-2 coding standard. It's become typical to structure doc blocks with:
```
/**
 *
 */
```

**References**: 
- https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#variable-types

File: php/add-subscriber-to-list.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+   * Typical use case is that this code would get run by an .ajax() jQuery call or possibly a form action
+   * The live data you need will get transferred via the global $_POST variable
+   * That data must be put into an array with keys that match the mailchimp endpoints, check the above link for those
+   * You also need to include your API key and list ID for this to work.
+   * You'll just have to go get those and type them in here, see README.md
+   * ================
+   */
+
+  // Set API Key and list ID to add a subscriber
+  $api_key = 'your-api-key-here';
```
Comment (deezone) [inline php/add-subscriber-to-list.php:17]: Consider using constants as they should not change for the lifetime of the script. Example:
```
define('API_KEY', 'your-api-key-here');
```
and access with:
```
curl_setopt($ch, CURLOPT_USERPWD, 'user:' . API_KEY);
```

File: php/add-subscriber-to-list.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  // It can vary, check the last 3 digits of your API key
+  $url = 'https://us5.api.mailchimp.com/3.0/lists/' . $list_id . '/members/';
+
+  // Encode data into a format that the add subscriber mailchimp end point is looking for
+  // Must include 'email_address' and 'status'
+  // Statuses: pending = they get an email; subscribed = they don't get an email
+  // Any custom fields go into the weirdly named 'merge_fields'
+  // More here: http://developer.mailchimp.com/documentation/mailchimp/reference/lists/members/#create-post_lists_list_id_members
+  $pfb_data = array(
+    'apikey'        => $apiKey,
```
Comment (kevnk) [inline php/add-subscriber-to-list.php]: `$apiKey` is an invalid variable. It should be:

```
   'apikey'        => $api_key,
```

**Comment by jserrao** [comment]
Good updates @kvenk + @DeeZone. Updating my PR accordingly.

**Pull request closed without merging**